src/fides/api/ops/util/connection_type.py

```diff
@@ -0,0 +1,144 @@
+from __future__ import annotations
+
+from typing import Any
+
+from fides.api.ops.common_exceptions import NoSuchConnectionTypeSecretSchemaError
+from fides.api.ops.models.connectionconfig import ConnectionType
+from fides.api.ops.schemas.connection_configuration import (
+    SaaSSchemaFactory,
+    secrets_schemas,
+)
+from fides.api.ops.schemas.connection_configuration.connection_config import (
+    ConnectionSystemTypeMap,
+    SystemType,
+)
+from fides.api.ops.schemas.saas.saas_config import SaaSConfig
+from fides.api.ops.service.connectors.saas.connector_registry_service import (
+    load_registry,
+    registry_file,
+)
+from fides.api.ops.util.saas_util import encode_file_contents, load_config
+
+
+def connection_type_secret_schema(*, connection_type: str) -> dict[str, Any]:
+    """Returns the secret fields that should be supplied to authenticate with a particular connection type.
+
+    Note that this does not return actual secrets, instead we return the *types* of
+    secret fields needed to authenticate.
+    """
+    connection_system_types: list[ConnectionSystemTypeMap] = get_connection_types()
+    if not any(item.identifier == connection_type for item in connection_system_types):
+        raise NoSuchConnectionTypeSecretSchemaError(
+            f"No connection type found with name '{connection_type}'."
+        )
+
+    if connection_type in [db_type.value for db_type in ConnectionType]:
+        return secrets_schemas[connection_type].schema()
+
+    config = SaaSConfig(**load_config(f"data/saas/config/{connection_type}_config.yml"))
```

Comment:
We should use the `ConnectorRegistry` so we can abstract out the location where the configs are loaded from.
```
ConnectorRegistry.get_connector_template(connection_type).config
```

Reply:
I tried making this update and it results in a bunch of mypy errors. I could fix some with modifications, but not all.

```py
    connector_template = ConnectorRegistry().get_connector_template(connection_type)
    if not connector_template:
        raise ValueError(f"Connection type {connection_type} template not found")
    config = connector_template.config
```

```console
src/fides/api/ops/util/connection_type.py:38: error: Missing named argument "__root__" for "ConnectorRegistry"  [call-arg]
        connector_template = ConnectorRegistry().get_connector_template(connection_type)
                             ^~~~~~~~~~~~~~~~~~~
src/fides/api/ops/util/connection_type.py:42: error: Argument 1 to "SaaSSchemaFactory" has incompatible type "str"; expected "SaaSConfig"  [arg-type]
        schema = SaaSSchemaFactory(config).get_saas_schema().schema()
                                   ^~~~~~
src/fides/api/ops/util/connection_type.py:46: error: "str" has no attribute "connector_params"  [attr-defined]
        order = [connector_param.name for connector_param in config.connector_params]
                                                             ^~~~~~~~~~~~~~~~~~~~~~~
src/fides/api/ops/util/connection_type.py:47: error: "str" has no attribute "external_references"  [attr-defined]
        if config.external_references:
           ^~~~~~~~~~~~~~~~~~~~~~~~~~
src/fides/api/ops/util/connection_type.py:51: error: "str" has no attribute "external_references"  [attr-defined]
                    for external_reference in config.external_references
```

Reply:
@sanders41, let's skip this suggestion for now then. The ConnectorRegistry needs some improvements and we can include this recommendation as part of those changes.